heat/core/signal.py

```diff
@@ -77,24 +88,23 @@ def convolve(a: DNDarray, v: DNDarray, mode: str = "full") -> DNDarray:
     a = a.astype(promoted_type)
     v = v.astype(promoted_type)
 
-    if v.is_distributed():
-        raise TypeError("Distributed filter weights are not supported")
     if len(a.shape) != 1 or len(v.shape) != 1:
         raise ValueError("Only 1-dimensional input DNDarrays are allowed")
-    if a.shape[0] <= v.shape[0]:
-        raise ValueError("Filter size must not be greater than or equal to signal size")
     if mode == "same" and v.shape[0] % 2 == 0:
         raise ValueError("Mode 'same' cannot be used with even-sized kernel")
 
+    if v.shape[0] > a.shape[0]:
+        a, v = v, a
+
     # compute halo size
-    halo_size = v.shape[0] // 2
+    halo_size = torch.max(v.lshape_map[:, 0]).item() // 2
```

Comment:
This seems to only partially solve the problem. It works for some configuration but for others it gives wrong results, for example on two processes:
import heat as ht 
import torch 
signal = ht.ones((7,), split=0)
weight = ht.ones((5,), split=0)
target_map = torch.zeros((weight.comm.size, 1), dtype=torch.int)
target_map[0, 0] = 4
target_map[1, 0] = 1
weight.redistribute_(target_map=target_map)
print(ht.convolve(signal, weight))

leads to:
DNDarray([1., 2., 3., 4., 5., 5., 5., 5., 5., 4., 3.], dtype=ht.float32, device=cpu:0, split=0)

whereas the correct result must be:
DNDarray([1., 2., 3., 4., 5., 5., 5., 4., 3., 2., 1.], dtype=ht.float32, device=cpu:0, split=0)

Reply:
yes, this convolve only supports weights which are balanced on all nodes, otherwise slicing method doesn't work due to more entries in `global_signal_filtered` as halo_size increased

Reply:
I see, however these cases should be caught in any case, either by an error message or by rebalancing as getting wrong results is dangerous.

Reply:
yes, even I think error message should be given. will do this asap.

Reply:
o.k., the code seems to be in a good shape now, from my side I would approve a PR. Nice work!

Reply:
Thanks @shahpratham , @krajsek , I'll approve and merge. Well done @shahpratham ! 👏🏼